src/lemlib/chassis/opcontrol.cpp

```diff
@@ -35,21 +35,33 @@ void Chassis::tank(int left, int right, bool disableDriveCurve) {
  * @param turn speed to turn. Takes an input from -127 to 127.
  * @param disableDriveCurve whether to disable the drive curve or not. If disabled, uses a linear curve with no
  * deadzone or minimum power
+ * @param desaturateBias how much to favor angular motion over lateral motion or vice versa when motors are
+ * saturated. A value of 0 fully prioritizes lateral motion, a value of 1 fully prioritizes angular motion
  */
-void Chassis::arcade(int throttle, int turn, bool disableDriveCurve) {
+void Chassis::arcade(int throttle, int turn, bool disableDriveCurve, float desaturateBias) {
     // use drive curves if they have not been disabled
     if (!disableDriveCurve) {
-        throttle = throttleCurve->curve(throttle);
-        turn = throttleCurve->curve(turn);
+        throttle = std::round(throttleCurve->curve(throttle));
+        turn = std::round(throttleCurve->curve(turn));
+    }
+    // desaturate motors based on joyBias
+    // desaturate motors based on joyBias
+    if (std::abs(throttle) + std::abs(turn) > 127) {
+        int oldThrottle = throttle;
+        int oldTurn = turn;
+        throttle *= (1 - desaturateBias * std::abs(oldTurn / 127.0));
+        turn *= (1 - (1 - desaturateBias) * std::abs(oldThrottle / 127.0));
+        // ensure the sum of the two values is equal to 127
+        // this check is necessary because of integer division
+        if (turn + throttle == 126) {
+            if (desaturateBias < 0.5) throttle += 1;
+            else turn += 1;
+        }
```

Comment:
this logic breaks if turn or throttle is negative. This should fix it
```suggestion
        if (std::abs(turn) + std::abs(throttle) == 126) {
            if (desaturateBias < 0.5) throttle += sgn(throttle);
            else turn += sgn(turn);
        }
```

Reply:
Does this work correctly when `desaturateBias` is exactly 0.5?

Reply:
it did in my testing